extensions/qxsck/var-and-list.js

```diff
@@ -5,292 +5,872 @@
+          {
+            opcode: "mapObject",
+            blockType: "command",
+            text: Scratch.translate(
+              "map object [OBJ] , keys to list [LIST], and values to list [LIST2]"
+            ),
+            arguments: {
+              OBJ: {
+                type: "string",
+                defaultValue: '{"name":"TruboWarp"}',
+              },
+              LIST: {
+                type: "string",
+                defaultValue: "list1",
+              },
+              LIST2: {
+                type: "string",
+                defaultValue: "list2",
+              },
+            },
+          },
+          {
+            opcode: "associateList",
+            blockType: "reporter",
+            text: Scratch.translate(
+              "associate list [LIST] to keys, and [LIST2] to values, if length is different, then [DO]"
+            ),
+            arguments: {
+              LIST: {
+                type: "string",
                 defaultValue: "list1",
               },
               LIST2: {
-                type: Scratch.ArgumentType.STRING,
+                type: "string",
                 defaultValue: "list2",
               },
+              DO: {
+                type: "string",
+                menu: "associateList.List",
+              },
+            },
+          },
+
+          {
+            opcode: "unSupportCompile",
+            blockType: "label",
+            text: Scratch.translate("⚠️Compilation not supported"),
+            arguments: {},
+          },
+
+          {
+            opcode: "forEach",
+            blockType: "conditional",
+            text: Scratch.translate(
+              "⚠️for each variable [VAR] from [LEFT] to [RIGHT]"
+            ),
+            arguments: {
+              VAR: {
+                type: "string",
+                defaultValue: "variable",
+              },
+              LEFT: {
+                type: "string",
+                defaultValue: "1",
+              },
+              RIGHT: {
+                type: "string",
+                defaultValue: "5",
+              },
+            },
+          },
+          {
+            opcode: "forEachList",
+            blockType: "conditional",
+            text: Scratch.translate(
+              "⚠️for each variable [VAR] in value of from [LEFT] to [RIGHT] in list [LIST]"
+            ),
+            arguments: {
+              VAR: {
+                type: "string",
+                defaultValue: "variable",
+              },
+              LIST: {
+                type: "string",
+                defaultValue: "list",
+              },
+              LEFT: {
+                type: "string",
+                defaultValue: "1",
+              },
+              RIGHT: {
+                type: "string",
+                defaultValue: "5",
+              },
             },
-            extensions: ["colours_data_lists"],
           },
         ],
+        menus: {
+          "openCaseSensitive.List": [
```

Comment:
Menus should NOT be formatted like this: this locks them into not accepting reporters. See the [documentation on menus](https://docs.turbowarp.org/development/extensions/inputs#static-menus) for more details.

Reply:
chang it now compromise compatibility